### PR TITLE
Introduce excludeSystemPortals query param

### PR DESCRIPTION
**Pull request by JeethJJ**
## Purpose
Introduce excludeSystemPortals parameter to exclude system portals such as console and myaccount (etc.) in GET applications api response. This is being introduced in order to resolve the pagination issue.

## To be merged after
https://github.com/wso2/carbon-identity-framework/pull/6026

### CODE REVIEW

**Review by shanggeeth — DISMISSED** [review]

File: components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/impl/ApplicationsApiServiceImpl.java [hunk 1]
```diff
@@ -67,12 +67,20 @@ public class ApplicationsApiServiceImpl implements ApplicationsApiService {
...
-                .getAllApplications(limit, offset, filter, sortOrder, sortBy, requiredAttributes);
+        return getAllApplications(limit, offset, filter, sortOrder, sortBy, requiredAttributes, true);
+    }
+
+    @Override
+    public Response getAllApplications(Integer limit, Integer offset, String filter, String sortOrder, String sortBy,
+                                       String requiredAttributes, Boolean excludeSystemPortals) {
+
+        ApplicationListResponse listResponse = applicationManagementService.getAllApplications(limit, offset, filter,
+                sortOrder, sortBy, requiredAttributes, Boolean.parseBoolean(String.valueOf(excludeSystemPortals)));
```
Comment (janakamarasena) [inline components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/impl/ApplicationsApiServiceImpl.java:83]: Can use Boolean.TRUE.equals(excludeSystemPortals)
```suggestion
                sortOrder, sortBy, requiredAttributes, Boolean.TRUE.equals(excludeSystemPortals));
```

Comment (JeethJJ) [reply components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/impl/ApplicationsApiServiceImpl.java]: Addressed with https://github.com/wso2/identity-api-server/pull/711/commits/0e01f11829d4f0c5baf9cffd6a422a5e8bdb2150

File: components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/impl/ApplicationsApiServiceImpl.java [hunk 2]
```diff
@@ -67,12 +67,20 @@ public class ApplicationsApiServiceImpl implements ApplicationsApiService {
...
     private ServerApplicationSharingService applicationSharingService;
 
+    @Deprecated
     @Override
     public Response getAllApplications(Integer limit, Integer offset, String filter, String sortOrder, String sortBy,
                                        String requiredAttributes) {
 
-        ApplicationListResponse listResponse = applicationManagementService
-                .getAllApplications(limit, offset, filter, sortOrder, sortBy, requiredAttributes);
+        return getAllApplications(limit, offset, filter, sortOrder, sortBy, requiredAttributes, true);
```
Comment (janakamarasena) [inline components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/impl/ApplicationsApiServiceImpl.java:75]: Shouldn't this set as false for excludeSystemPortals?

Comment (JeethJJ) [reply components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/impl/ApplicationsApiServiceImpl.java]: Yes. Addressed with https://github.com/wso2/identity-api-server/pull/711/commits/1db3e049a17d85a6974525c93f50d27520ddb979

**Review by janakamarasena — APPROVED** [review]

**Pull request merged**